src/main/java/com/bobocode/svydovets/context/AnnotationConfigurationApplicationContext.java

```diff
@@ -1,29 +1,77 @@
 package com.bobocode.svydovets.context;
 
 import java.util.Map;
+import java.util.stream.Collectors;
 
+import org.apache.commons.lang3.StringUtils;
+
+import com.bobocode.svydovets.beans.factory.BeanFactory;
+import com.bobocode.svydovets.beans.factory.DefaultListableBeanFactory;
+import com.bobocode.svydovets.beans.scanner.ComponentBeanScanner;
 import com.bobocode.svydovets.exception.NoSuchBeanDefinitionException;
 import com.bobocode.svydovets.exception.NoUniqueBeanDefinitionException;
 
 public class AnnotationConfigurationApplicationContext implements ApplicationContext {
+    private Map<String, Object> nameToBean;
+    private final BeanFactory beanFactory;
+    private final ComponentBeanScanner componentScanner;
+
+    public AnnotationConfigurationApplicationContext(String packageName) {
+        this.beanFactory = new DefaultListableBeanFactory();
+        this.componentScanner = new ComponentBeanScanner();
+        this.scan(packageName);
+    }
+
+    public AnnotationConfigurationApplicationContext(BeanFactory beanFactory,
+                                                     ComponentBeanScanner componentScanner,
+                                                     String packageName) {
+        this.beanFactory = beanFactory;
+        this.componentScanner = componentScanner;
+        this.scan(packageName);
+    }
 
     @Override
     public <T> T getBean(Class<T> beanType) throws NoSuchBeanDefinitionException, NoUniqueBeanDefinitionException {
-        return null;
+        var beans = nameToBean.values().stream()
```

Comment:
Could you please add tests for get methods?

Reply:
I'll approve and create a separate task for unit test AnnotationConfigurationApplicationContext.java.